cmd/containerd-shim-runhcs-v1/pod.go

```diff
@@ -115,16 +126,23 @@ func createPod(ctx context.Context, events publisher, req *task.CreateTaskReques
 			}
 			layers[layersLen-1] = vmPath
 			wopts.LayerFolders = layers
-
-			parent, err = uvm.CreateWCOW(ctx, wopts)
+			if utc != nil {
+				parent, err = uvm.CloneWCOW(ctx, wopts, utc)
+			} else {
+				parent, err = uvm.CreateWCOW(ctx, wopts)
+			}
```

Comment:
personal style choice here: can we move this if/else logic inside of like `CreateWCOW` which could call `cloneWCOW` or `createWCOW` internally?

Reply:
after reading more through the code, I think this would be a good idea. There is a lot of overlap between CreateWCOW and CloneWCOW.

Reply:
I just wanted to keep the cloning path separate from the normal code path to avoid making existing functions complex. But as of now these functions don't get very complex after addition of clone related changes. In the future if we make more changes to the cloning code path then we can refactor if required.